eos/b-decays/b-to-gamma-l-nu_TEST.cc

```diff
@@ -0,0 +1,134 @@
+/* vim: set sw=4 sts=4 et foldmethod=syntax : */
+
+/*
+ * Copyright (c) 2022 Philip Lüghausen
+ *
+ * This file is part of the EOS project. EOS is free software;
+ * you can redistribute it and/or modify it under the terms of the GNU General
+ * Public License version 2, as published by the Free Software Foundation.
+ *
+ * EOS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59 Temple
+ * Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#include <test/test.hh>
+#include <eos/b-decays/b-to-gamma-l-nu.hh>
+#include <eos/maths/integrate.hh>
+#include <eos/maths/integrate-impl.hh>
+#include <eos/observable.hh>
+
+using namespace test;
+using namespace eos;
+
+class BToGammaLeptonNeutrinoTest :
+    public TestCase
+{
+    public:
+        BToGammaLeptonNeutrinoTest() :
+            TestCase("b_to_gamma-l_nu_test")
+        {
+        }
+
+        virtual void run() const
+        {
+            static const double eps = 1e-6;
+
+            Parameters p = Parameters::Defaults();
+
+            // Values from [BBJW:2018A]
+            p["B_u::mu_0@FLvD2022"]  = 1.5;
+            p["B->gamma::mu@FLvD2022QCDF"]  = 1.5;
+            p["B->gamma::mu_h1@FLvD2022QCDF"]  = 4.7;
+            p["B->gamma::mu_h2@FLvD2022QCDF"]  = 4.5;
+            p["B->gamma::s_0@FLvD2022QCDF"] = 1.59;
+            p["B->gamma::M^2@FLvD2022QCDF"] = 1.35;
+            p["decay-constant::B_u"] = 0.192;
+            p["mass::B_u"] = 5.27929;
+            p["mass::b(MSbar)"] = 4.246057986045; // fix m_b_pole to 4.8
```

Comment:
Be careful here: EOS uses by default 3-loop relations between the MSbar and the pole scheme. The pole scheme is ill-defined, and does not converge toward a consistent value as a function of the loop order. If you use a 1-loop expression in your code, m_b_pole should be calculated from m_b_MSbar at one-loop only.